### ISSUE TITLE
[css2] Switching to single-page broke existing URLs

### ISSUE THREAD

**Issue by SimonSapin** [w3c/csswg-drafts#5330]
It looks like https://drafts.csswg.org/css2/ was switched to a single-page. This broke many links that exist in the wild to URLs like https://drafts.csswg.org/css2/visudet.html#propdef-vertical-align. Can we add redirects?

Presumably https://www.w3.org/TR/CSS2/ will have the same issue when it’s updated.

**Comment by gsnedders** [comment]
@plinss is there any reason why the `.htaccess` isn't successfully rewriting this?

**Comment by plinss** [comment]
I’ll take a look shortly. 

This raises another concern that all the 2.1 test suite links are now broken as well. And redirects won’t fix that.

**Comment by SimonSapin** [comment]
This is perhaps to discuss in a separate issue, but I wonder if switching CSS 2 to a single page is really worth the trouble

**Comment by plinss** [comment]
The redirects are in place now. I'm keeping this issue open due to the test links being broken.

Personally, I'm glad for the Bikeshed conversion, and have no issue with a single page spec, but I'd like to see a multipage version with equivalent URLs to the old version.

I accept that's currently blocked by Bikeshed's lack of multipage support.

If we can't generate a multi-page version, then we need to either edit all the CSS2.1 tests links (which would not only be a large amount of work but would also take work to avoid invalidating old test results), or we need to build a system to translate the old URLs into the current form for the test suite build system and Shepherd.

**Comment by gsnedders** [comment]
> we need to build a system to translate the old URLs into the current form for the test suite build system and Shepherd

Where does Shepherd read from? It reads from one of the built products, right? I can certainly do something in the build system on Monday to get it rewriting URLs (draft URLs obviously need changing, but what about TR? do we want to use dated URLs or just change it when we republish?).

**Comment by plinss** [comment]
Shepherd parses the built specs directly. We don't want to change the Shepherd data because Bikeshed and Respec use it to generate links to CSS2 and those need to be correct.

The build system could get a translation layer so it at least recognizes old CSS2 links and associates them to the appropriate spec sections. The test harness also needs to deal with those changes so the test suite annotations work for CSS2. (It's possible the build translation can simply carry over to the harness, I'd have to check the code.)
